rig/place_and_route/place/breadth_first.py

```diff
@@ -0,0 +1,59 @@
+"""A greedy vertex-ordering heuristic for a sequential placer."""
+
+from collections import deque, defaultdict
+
+from rig.place_and_route.place.sequential import place as sequential_place
+
+
+def breadth_first_vertex_order(vertices_resources, nets):
+    """A generator which iterates over a set of vertices in a breadth-first
+    order in terms of connectivity.
+
+    For use as a vertex ordering for the sequential placer.
+    """
+    # Special case: no vertices, just stop immediately
+    if len(vertices_resources) == 0:
+        return
+
+    # Enumerate the set of nets attached to each vertex
+    vertex_neighbours = defaultdict(set)
+    for net in nets:
+        vertex_neighbours[net.source].update(net)
```

Comment:
Other option: `vertex_neighbours[net.source].update(iter(net))` just to make it explicit?